packages/contracts-bedrock/test/safe/DeputyPauseModule.t.sol

```diff
@@ -0,0 +1,675 @@
+        foundationSafeInstance = _setupSafe(keys2, 10);
+
+        // Set the Security Council Safe as the Guardian of the SuperchainConfig.
+        vm.store(
+            address(superchainConfig),
+            superchainConfig.GUARDIAN_SLOT(),
+            bytes32(uint256(uint160(address(securityCouncilSafeInstance.safe))))
+        );
+
+        // Create a DeputyGuardianModule and set the Foundation Safe as the Deputy Guardian.
+        deputyGuardianModule = IDeputyGuardianModule(
+            DeployUtils.create1({
+                _name: "DeputyGuardianModule",
+                _args: DeployUtils.encodeConstructor(
+                    abi.encodeCall(
+                        IDeputyGuardianModule.__constructor__,
+                        (securityCouncilSafeInstance.safe, superchainConfig, address(foundationSafeInstance.safe))
+                    )
+                )
+            })
+        );
+
+        // Enable the DeputyGuardianModule on the Security Council Safe.
+        securityCouncilSafeInstance.enableModule(address(deputyGuardianModule));
+
+        // Create the deputy for the DeputyPauseModule.
+        (deputy, deputyKey) = makeAddrAndKey("deputy");
+
+        // Create the deputy auth signature.
+        deputyAuthSignature = makeAuthSignature(getNextContract(), deputyKey, deputy);
+
+        // Create the DeputyPauseModule.
+        deputyPauseModule = IDeputyPauseModule(
+            DeployUtils.create1({
+                _name: "DeputyPauseModule",
+                _args: DeployUtils.encodeConstructor(
+                    abi.encodeCall(
+                        IDeputyPauseModule.__constructor__,
+                        (foundationSafeInstance.safe, deputyGuardianModule, superchainConfig, deputy, deputyAuthSignature)
+                    )
+                )
+            })
+        );
+
+        // Enable the DeputyPauseModule on the Foundation Safe.
+        foundationSafeInstance.enableModule(address(deputyPauseModule));
+    }
+
+    /// @notice Generates a signature to authenticate as the deputy.
+    /// @param _verifyingContract The verifying contract.
+    /// @param _privateKey The private key to use to sign the message.
+    /// @param _deputy The deputy to authenticate as.
+    /// @return Generated signature.
+    function makeAuthSignature(
+        address _verifyingContract,
+        uint256 _privateKey,
+        address _deputy
+    )
+        internal
+        view
+        returns (bytes memory)
+    {
+        return makeAuthSignature(block.chainid, _verifyingContract, _privateKey, _deputy);
+    }
+
+    /// @notice Generates a signature to authenticate as the deputy.
+    /// @param _chainId Chain ID to use for the domain separator.
+    /// @param _verifyingContract The verifying contract.
+    /// @param _privateKey The private key to use to sign the message.
+    /// @param _deputy The deputy to authenticate as.
+    /// @return Generated signature.
+    function makeAuthSignature(
```

Comment:
I don't see any usages of this 4 arg variant except within the 3 arg variant above.

Reply:
`testFuzz_constructor_wrongChainId_reverts` was supposed to do this, that test has a bug. I just fixed it.